sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import java.util.Set;
+
+/**
+ * The log configurations for HTTP messages.
+ */
+public class HttpLogOptions {
+    public HttpLogOptions(final HttpLogDetailLevel logLevel) {
```

Comment:
I would make this a no-arg constructor and have a fluent API. Have each set method return 'this', so that it can be chained.